# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1290129292 18000
#      Thu Nov 18 20:14:52 2010 -0500
# Node ID 521f2bb7c443b19326f4f970427b217cd8e460f4
# Parent  da314c58f0b9bc9c4e05e59118ee803fe908c470
text.m: Ensure text position property is set after units property.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,12 @@
 2010-11-18  Ben Abbott <bpabbott@mac.com>
 
+	* plot/text.m: Ensure text position property is set after units
+	property.
 	* plot/__go_draw_axes__.m: Fix bug setting zscale to yscale (31693).
 
 2010-11-18  Christos Dimitrakakis  <christos.dimitrakakis@gmail.com>
 
 	* statistics/distributions/betapdf.m: Use lgamma to compute
 	normalising constant in log space in order to handle large
 	parameters a and b.  Ensure correct values at x == 0 or x == 1.
 
diff --git a/scripts/plot/text.m b/scripts/plot/text.m
--- a/scripts/plot/text.m
+++ b/scripts/plot/text.m
@@ -71,25 +71,25 @@ function h = text (varargin)
     if (nx == ny && nx == nz)
       pos = [x(:), y(:), z(:)];
       ca = gca ();
       tmp = zeros (n, 1);
       if (n == 1)
         label = label{1};
         for i = 1:nx
           tmp(i) = __go_text__ (ca, "string", label,
-                                "position", pos(i,:),
-                                varargin{:});
+                                varargin{:},
+                                "position", pos(i,:));
         endfor
         __request_drawnow__ ();
       elseif (n == nx)
         for i = 1:nx
           tmp(i) = __go_text__ (ca, "string", label{i},
-                                "position", pos(i,:),
-                                varargin{:});
+                                varargin{:},
+                                "position", pos(i,:));
         endfor
         __request_drawnow__ ();
       else
         error ("text: dimension mismatch for coordinates and label");
       endif
     else
       error ("text: dimension mismatch for coordinates");
     endif
